doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,271 @@
+# Listing Content and Locations
+
+Starting from Platform 2015.07, listing content and locations in a project is
+much easier, using a couple standardized mechanisms.
+
+Predefined, named queries can be defined in any bundle. They can then be used from twig, using a
+[custom controller](#query_controller), or from PHP code using the [QueryType Registry](#fixme). The controller supports
+[template overriding](#template_override), making it easy to use different templates for different calls.
+
+## <a name="query_controller"></a> The query controller
+A new controller is added : `ez_query`.
+
+It has two actions: `locationQuery`, and `contentQuery`.
+
+Both requires two arguments:
+- `queryName`, the name of the QueryType to run. Example: `AcmeBundle:LatestArticles'
+- `viewType`, similar to what `viewContent` and `viewLocation` expect. Examples: `list`, `tree`, `full_list`...
+
+It also supports a `parameters` hash. It will be passed on to the QueryType when building the Query, and made available
```

Comment:
'parameters' seems to refer to 'queryTypeParameters' below.

besides, 1. a bit unclear why (query)Type is used one place but not the other, 2. given the action name contains the name query, I think we can simplify:

so maybe just:
```jinja
    {
        'name': 'AcmeBundle:LatestContent',
        'viewType': 'list',
        'parameters': {'type': 'article'}
    }
```

Not a strong suggestion, but was just a first impression on this code snippet.